default.nix

```diff
@@ -614,17 +637,41 @@ in
           serviceConfig.Type = "oneshot";
           serviceConfig.RemainAfterExit = true;
           path = [ pkgs.iproute2 ];
-          script = ''
-            ip link show dev "${interface}" >/dev/null 2>&1 && ip link del "${interface}" || true
-            echo "Adding bridge ${interface}..."
-            ip link add name "${interface}" type bridge
-            ip link set "${interface}" up
-            echo -n > "/run/${interface}.interfaces"
-            ${lib.concatMapStrings (i: ''
-              ip link set "${i}" master "${interface}" up && echo "${i} " >> "/run/${interface}.interfaces" || true
-            '') value}
-            ip link set "${interface}" up
-          '';
+          script =
+            let
+              vlans = if lib.hasAttr interface config.router.bridges then config.router.bridges.${interface}.vlans else {};
+              vlan_filtering = vlans != {}; 
+            in
+            ''
+              ip link show dev "${interface}" >/dev/null 2>&1 && ip link del "${interface}" || true
+              echo "Adding bridge ${interface}..."
+              ip link add name "${interface}" type bridge
+
+              ${
+                lib.optionalString vlan_filtering ''
+                  ip link set dev "${interface}" type bridge vlan_filtering 1
+                  bridge vlan del dev "${interface}" vid 1 self
+                ''
+              }
+
+              ip link set "${interface}" up
+              echo -n > "/run/${interface}.interfaces"
+              ${lib.concatMapStrings (i: 
+              let 
+                vid_commands = lib.optionalString (lib.hasAttr i vlans) (lib.concatMapStrings (x: 
+                ''
+                  bridge vlan add dev "${i}" vid ${builtins.toString x.vid} ${lib.optionalString x.untagged "pvid untagged"}
```

Comment:
`pvid untagged` is two options. Excerpt from manpages:

> **pvid** - the vlan specified is to be considered a PVID at ingress.  Any untagged frames will be assigned to this VLAN.
> **untagged** - the vlan specified is to be treated as untagged on egress.

Is there a reason they are together here?

Reply:
For me it was more intuitive that if I have untagged frames coming in on the interface that are forced to a specific vlan, outgoing packets with this vlan id should be untagged as well. Could definitely be better documented. Not sure if another option is needed.